reactor-core/src/main/java/reactor/core/publisher/SinkManyBestEffort.java

```diff
@@ -0,0 +1,385 @@
+
+		if (subs == EMPTY) {
+			return Emission.FAIL_ZERO_SUBSCRIBER;
+		}
+		if (subs == TERMINATED) {
+			return Emission.FAIL_TERMINATED;
+		}
+
+		if (allOrNothing) {
+			long commonRequest = Long.MAX_VALUE;
+			for (Inner<T> sub : subs) {
+				long subRequest = sub.requested;
+				if (sub.isCancelled()) {
+					continue;
+				}
+				if (subRequest < commonRequest) {
+					commonRequest = subRequest;
+				}
+			}
+			if (commonRequest == 0) {
+				return Emission.FAIL_OVERFLOW;
+			}
+		}
+
+		int expectedEmitted = subs.length;
+		int emittedCount = 0;
+		int cancelledCount = 0;
+		for (Inner<T> sub : subs) {
+			if (sub.isCancelled()) {
+				cancelledCount++;
+				continue;
+			}
+			if (sub.tryEmitNext(t)) {
+				emittedCount++;
+			}
+			else if (sub.isCancelled()) {
+				cancelledCount++;
+			}
+		}
+		if (cancelledCount == expectedEmitted) {
+			return Emission.FAIL_ZERO_SUBSCRIBER; //TODO should it be FAIL_CANCELLED ? but technically the sink would still be usable
+		}
+		else if (cancelledCount + emittedCount == expectedEmitted) {
+			return Emission.OK;
+		}
+		else {
+			return Emission.FAIL_OVERFLOW;
+		}
+	}
+
+	@Override
+	public Emission tryEmitComplete() {
+		@SuppressWarnings("unchecked")
+		Inner<T>[] subs = SUBSCRIBERS.getAndSet(this, TERMINATED);
+
+		if (subs == TERMINATED) {
+			return Emission.FAIL_TERMINATED;
+		}
+
+		for (Inner<?> s : subs) {
+			s.emitComplete();
+		}
+		return Emission.OK;
+	}
+
+	@Override
+	public Emission tryEmitError(Throwable error) {
+		Objects.requireNonNull(error, "tryEmitError(null) is forbidden");
+		@SuppressWarnings("unchecked")
+		Inner<T>[] subs = SUBSCRIBERS.getAndSet(this, TERMINATED);
+
+		if (subs == TERMINATED) {
+			return Emission.FAIL_TERMINATED;
+		}
+
+		this.error = error;
+
+		for (Inner<?> s : subs) {
+			s.emitError(error);
+		}
+		return Emission.OK;
+	}
+
+	@Override
+	public void emitComplete() {
+		//no particular error condition handling for onComplete
+		@SuppressWarnings("unused")
+		Emission emission = tryEmitComplete();
+	}
+
+	@Override
+	public void emitError(Throwable error) {
+		Emission result = tryEmitError(error);
+		if (result == Emission.FAIL_TERMINATED) {
+			Operators.onErrorDroppedMulticast(error, subscribers);
+		}
+	}
+
+	@Override
+	public void emitNext(T value) {
+		switch(tryEmitNext(value)) {
+			case FAIL_ZERO_SUBSCRIBER:
+				//we want to "discard" without rendering the sink terminated.
+				// effectively NO-OP cause there's no subscriber, so no context :(
+				break;
+			case FAIL_OVERFLOW:
+				Operators.onDiscard(value, currentContext());
+				//the emitError will onErrorDropped if already terminated
+				emitError(Exceptions.failWithOverflow("Backpressure overflow during Sinks.Many#emitNext"));
+				break;
+			case FAIL_CANCELLED:
+				Operators.onDiscard(value, currentContext());
+				break;
+			case FAIL_TERMINATED:
+				Operators.onNextDroppedMulticast(value, subscribers);
+				break;
+			case OK:
+				break;
+			case FAIL_NON_SERIALIZED:
```

Comment:
```suggestion
			default:
```

Reply:
I have a slight preference for not using `default` and having full explicit enum coverage, that way if we add an enum value the IDE can be configured to warn us. at least in `tryEmitNext`, which is the most likely to make use of most enums.

Reply:
(Marked as not resolved, since... it is not resolved yet ;))

Unless we make our _build_ fail when we forget to handle a new case, I'd say relying on IDE isn't a good idea. Also, not every team member/contributor has this option configured (I don't)

Reply:
updated